google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/mutaterows/MutateRowsBatchingDescriptor.java

```diff
@@ -46,7 +46,11 @@ public BatchingRequestBuilder<RowMutationEntry, BulkMutation> newRequestBuilder(
   }
 
   @Override
-  public void splitResponse(Void response, List<BatchEntry<RowMutationEntry, Void>> entries) {
+  public void splitResponse(
+      MutateRowsAttemptResult response, List<BatchEntry<RowMutationEntry, Void>> entries) {
+    for (FailedMutation mutation : response.failedMutations) {
+      entries.get(mutation.getIndex()).getResultFuture().setException(mutation.getError());
+    }
```

Comment:
please add a comment explaining the logic here....its not obvious that ordering matters here and that you are depending on the future being able to be rsesolved once

Reply:
Done